### ISSUE TITLE
cy.screenshot not respecting the viewport size on Firefox.

### ISSUE THREAD

**Issue by aayknija** [cypress-io/cypress#15481]
Using: Cypress package version: 6.6.0,  OS-Windows 10 , Firefox -86.0.1 (64-bit)

As per the documentation, I have added the below code snippet in plugins/index.js still it is not working as expected. The expectation is to take the screenshot in provided resolution.

Command: **cypress run --browser firefox --headless** 

```js
on('before:browser:launch', (browser = {}, launchOptions) => {
    if (browser.name === 'firefox') {
      launchOptions.args.push('--width=1680');
      launchOptions.args.push('--height=1024');
      return launchOptions;
    }
  });
```
  
  P.S: It works perfectly on chrome with the below code snippet in plugins/index.js

 ```js
on('before:browser:launch', (browser = {}, launchOptions) => {
    if (browser.name === 'chrome') {
      launchOptions.args.push('--disable-dev-shm-usage');
      launchOptions.args.push('--window-size=1680,1024');
      launchOptions.args.push('--disable-gpu');
      return launchOptions;
    }
  });
```
  
  I tried adding `--disable-dev-shm-usage` and `--disable-gpu` for firefox also but it did not work.

**Comment by Bkucera** [comment]
@aayknija can you provide what screenshot size you're getting with those options in firefox? It's possible the window size for firefox includes the ~74 pixels of titlebar/menu, so the screenshots may be 1680x950. If that's the case you can try adding 74px to the height argument. We haven't normalized this for users since it's unclear how stable it would be. 

```js
   if (browser.name === 'firefox') {
      launchOptions.args.push('--width=1680');
      launchOptions.args.push('--height=1098');
      return launchOptions;
    }
```

**Comment by aayknija** [comment]
@Bkucera Apologies for the delayed response. I did try the above option as well. The issue was it was not respecting the width and height provided in launch options. I was getting an image with a resolution of 1680x950 and size did vary in the docker image.

Finally, after some google search, I found one solution where I have to define **MOZ_HEADLESS_WIDTH** and **MOZ_HEADLESS_HEIGHT** in the environment variable and that worked for me.
